tiledb/sm/misc/parallel_functions.h

```diff
@@ -270,17 +221,7 @@ Status parallel_for(
   }
 
   // Wait for all instances of `execute_subrange` to complete.
-  // This is ignoring the wait status as we use failed_exception for propagating
-  // the tasks exceptions.
-  std::ignore = tp->wait_all(tasks);
-
-  if (failed_exception.has_value()) {
-    std::rethrow_exception(failed_exception.value());
-  }
-  if (failed_status.has_value()) {
-    return failed_status.value();
-  }
-  return Status{};  // otherwise return OK
+  return tp->wait_all(tasks);
```

Comment:
This is much better than the previous error-handling code in `parallel_for` (which was somewhat broken, IMO).  However, in the current implementation of `ThreadPool::wait_all()`, any exceptions in the tasks will be caught and a corresponding status will be put in the `status` vector.  `ThreadPool::wait_all` will return the value of the first non-Ok element in the `status` vector filled by `ThreadPool::wait_all_status`.  The code above will similarly return the first non-Ok return value from the set of `execute_subrange` invocations. Important information about multiple failures (and the kind of failures) will be lost.

In a future story/PR, we should define a new wait function for the thread pool that throws a "compound" exception containing information about all the waited-on tasks that threw exceptions.  (Or, just redefine `wait_all` -- though we might want to rename `wait_all` to something like `sync_wait_all` just to call out its behavior).